app/src/main/java/ru/vssemikoz/newsfeed/newsfeed/NewsFeedPresenter.java

```diff
@@ -43,17 +49,43 @@ public NewsFeedPresenter() {
     }
 
     @Override
-    public void start() {
+    public void subscribe() {
         Log.d(TAG, "start: " + mainApplication);
         Category.resolveCategory(mainApplication, Arrays.asList(Category.values()));
         initStartValues();
         updateActualNews();
     }
 
+    @Override
+    public void unsubscribe() {
+        disposables.clear();
+    }
+
     @Override
     public void updateActualNews() {
-        view.showProgressBar();
-        updateNewsStorage();
+        NewsFeedParams params = new NewsFeedParams(new Filter(category, showOnlyFavorite));
+        Disposable subscription =
+                updateStorageUseCase.run(params)
+                        .doOnSubscribe(disposable -> view.showProgressBar())
+                        .subscribeOn(Schedulers.computation())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(
+                                updatedNews -> { /* Emit */
+                                    getNewsFromStorage();
```

Comment:
This stuff should be removed from here. The idea of this chain is to be launched to download data, save it and pass to the UI. You can use a `flatMap` operator to save data in your use cases or repository so you'll later get the data from the storage somewhere upper in the chain and will use the result list here to update the UI. As for now, it's not used at all and you have two separate use cases and network calls to do the same thing.